diff --git a/main.js b/main.js
index febfc0769239db5f0c95a6f80620241b9866c298..ce7e34dc3c7bd2355ef56a984c2cea1af99f64f1 100755
--- a/main.js
+++ b/main.js
@@ -32031,7 +32031,14 @@ var _PackageJson = class _PackageJson extends Class3("PackageJson")({
     executableFiles: optional(Array$(String$))
   })),
   license: String$,
-  author: optional(String$),
+  author: optional(Union2(
+    String$,
+    Struct({
+      name: String$,
+      email: String$,
+      organization: optional(Boolean$)
+    })
+  )),
   repository: Union2(
     String$,
     Struct({
